apps/shipments/serializers/shipment_overview.py

```diff
@@ -53,10 +53,27 @@ def validate_in_bbox(self, in_bbox):
                 raise exceptions.ValidationError('Invalid geo box, make sure that: '
                                                  'in_bbox[1] < in_bbox[3] and in_bbox[2] < in_bbox[4].')
 
-            return ','.join([c.strip() for c in in_bbox.split(',')])
+            return in_bbox_num
         return None
 
-    def validate_state(self, state):
-        if state.name == TransitState.AWAITING_PICKUP.name:
-            raise exceptions.ValidationError(f'[{state.name}] is an invalid state value!')
-        return state.name
+
+class ShipmentLocationSerializer(serializers.ModelSerializer):
+    location = serializers.SerializerMethodField()
```

Comment:
Why not use a GeoJSON serializer field here instead of a methodfield that manually serializes the geojson field?

Reply:
I'm not sure I understand what you mean here. A GeoJson serializer alone won't work on a subclass of `ModelSerializer` or you did mean something else?
Also here we are serializing in theory a list of "Shipment" objects tagged each with a location. I'm not sure `FeatureCollection` will be helpful here since we will loose the single "shipment/device" info  see below sample response:

```
"data": [
        {
            "type": "TrackingData",
            "id": "12649be1-04b9-447e-9ea7-bb72076b7949",
            "attributes": {
                "location": {
                    "type": "Feature",
                    "properties": {
                        "source": "gps",
                        "uncertainty": 64,
                        "time": "2020-02-04T19:29:14.014Z"
                    },
                    "geometry": {
                        "type": "Point",
                        "coordinates": [
                            -85.46946468273921,
                            34.48124935925382
                        ]
                    }
                }
            },
            "relationships": {
                "device": {
                    "data": {
                        "type": "Device",
                        "id": "ef5f3cea-291e-4160-bb69-f5f967b9be3f"
                    }
                },
                "shipment": {
                    "data": {
                        "type": "Shipment",
                        "id": "a86a53af-2dd7-4a86-9658-954fe728984e"
                    }
                }
            }
        }
    ]
```

Reply:
I see. If you are just wanting to render our geometry point field, I recommend using the existing drf-gis field for this: https://github.com/openwisp/django-rest-framework-gis/blob/master/rest_framework_gis/fields.py#L14

Also, I would rename 'location' to 'point' which is what the geometry is called on the model.

And if you don't need a FeatureCollection, then I don't understand the purpose of the MultiFeatureTrackingDataSerializer.

Reply:
In order to use `rest_framework_gis.field.GeometryField`, we need to use `GeoFeatureModelSerializer` which is incompatible with our `rest_framework_json_api`.
Instead I kept the `MethodSerializer` in place and make use of the refactored `SingleFeatureTrackingDataSerializer`